lib/pbench/server/database/alembic/versions/1a91bc68d6de_update_api_key.py

```diff
@@ -1,29 +1,33 @@
 """Update api_key table with "id" as primary_key
 
-Revision ID: e695e86f722a
+
+Revision ID: 1a91bc68d6de
 Revises: 5679217a62bb
-Create Date: 2023-04-27 18:59:31.914683
+Create Date: 2023-05-03 09:50:29.609672
 
 """
 from alembic import op
 import sqlalchemy as sa
 
 # revision identifiers, used by Alembic.
-revision = "e695e86f722a"
+revision = "1a91bc68d6de"
 down_revision = "5679217a62bb"
 branch_labels = None
 depends_on = None
 
 
 def upgrade() -> None:
+    op.drop_constraint("api_keys_pkey", "api_keys", type_="primary")
     op.add_column(
         "api_keys", sa.Column("id", sa.Integer(), autoincrement=True, nullable=False)
     )
-    op.add_column("api_keys", sa.Column("name", sa.String(length=128), nullable=False))
-    op.create_unique_constraint(None, "api_keys", ["api_key"])
+    op.add_column("api_keys", sa.Column("name", sa.String(length=128), nullable=True))
```

Comment:
Again, if we don't ever use the value `None`, there's no point in making it `nullable`. Conversely, if it's nullable, let's allow a null value rather than forcing that case to an empty string. I'm not sure there's any great value in being able to distinguish between `None` and `""` at the client level, and I can't really imagine the dashboard displaying them differently anyway, but I'd prefer that "none specified" means `None`.

Reply:
@siddardh-ra : regarding the autoincrement problem you saw, it seems that alembic's `autoincrement` flag on `add_column` only works for MySQL. That's mind-boggling, really, and I wasn't able to find a concise and portable way to fix it, but here's an "old-school SQL" version of the `upgrade` that seems to work correctly:

```python
def upgrade() -> None:
    op.drop_constraint("api_keys_pkey", "api_keys", type_="primary")
    op.execute("ALTER TABLE api_keys ADD COLUMN id SERIAL PRIMARY KEY")
    op.add_column("api_keys", sa.Column("name", sa.String(length=128), nullable=True))
    op.create_unique_constraint("api_keys_api_key_key", "api_keys", ["api_key"])
```

and results in the following table description:

```
pbenchcontainer=> \d api_keys
                                       Table "public.api_keys"
 Column  |            Type             | Collation | Nullable |               Default                
---------+-----------------------------+-----------+----------+--------------------------------------
 api_key | character varying(500)      |           | not null | 
 created | timestamp without time zone |           | not null | 
 user_id | character varying           |           | not null | 
 id      | integer                     |           | not null | nextval('api_keys_id_seq'::regclass)
 name    | character varying(128)      |           | not null | 
Indexes:
    "api_keys_pkey" PRIMARY KEY, btree (id)
    "api_keys_api_key_key" UNIQUE CONSTRAINT, btree (api_key)
Foreign-key constraints:
    "api_keys_user_id_fkey" FOREIGN KEY (user_id) REFERENCES users(id)
```